.github/workflows/test.yml

```diff
@@ -16,7 +16,7 @@ jobs:
         uses: ruby/setup-ruby@v1
         with:
           ruby-version: 2.5.1
-          bundler-cache: true
+          bundler-cache: false
```

Comment:
How come you needed to disable this? There is a `cache_version` attribute you can bump if the [cache is for some reason corrupted](https://github.com/ruby/setup-ruby#dealing-with-a-corrupted-cache). 

But more importantly, maybe it makes sense to set up a build matrix to test LHM against multiple Ruby and Rails versions? GitHub has some [documentation on this], and [this would be an example] of implementing this for a rails gem. Over there, I used [appraisal] to generate the different Gemfiles. It also gives you some tasks that will let you run tests against all the different Gemfiles locally.

[documentation on this]: https://docs.github.com/en/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
[this would be an example]: https://github.com/check24-profis/shared-cursor-pager/blob/main/.github/workflows/ci.yml
[appraisal]: https://github.com/thoughtbot/appraisal

Reply:
Update on the cache: Running the test suite with `bundler-cache: true` fails in the pipeline due to `ArgumentError: Trying to register Bundler::GemfileError for status code 4 but Bundler::GemfileError is already registered`.
It seems like the test workflow doesn't like appraisal and using multiple `gemfiles`. There's no real documented cases on the internet about this error in our context, but using `bundler-cache: false` allows the workflow to pass without any problems.

Reply:
Take a look at the example I posted in the first message in this thread. On that project, I was only using appraisals to manage the various gemfiles and for local testing. In the GitHub action, I didn't use it but instead used a build matrix. The action will spawn a separate job for each item in the matrix and run those jobs in parallel. So it shouldn't make the pipeline significantly longer.